contracts/origination/OriginationControllerSTIRFRY.sol

```diff
@@ -0,0 +1,268 @@
+            )
+        );
+        if(stirfryPairs[currencyHash] != true)
+            revert OCS_InvalidStirfryPair(loanTerms.payableCurrency, stirfryData.vaultedCurrency);
+
+        // verify the vaulted currency amounts
+        if(loanTerms.principal * stirfryData.payableToVaultedCurrencyRatio != stirfryData.lenderVaultedCurrencyAmount)
+            revert OCS_InvalidPrincipalAmounts(
+                loanTerms.principal,
+                stirfryData.payableToVaultedCurrencyRatio,
+                stirfryData.lenderVaultedCurrencyAmount
+            );
+
+        // verify that the vault holds the lenderVaultedCurrencyAmount
+        uint256 vaultAmount = IERC20(stirfryData.vaultedCurrency).balanceOf(vaultAddress);
+        if(vaultAmount != stirfryData.lenderVaultedCurrencyAmount)
+            revert OCS_InvalidVaultAmount(vaultAmount, stirfryData.lenderVaultedCurrencyAmount);
+
+        // calculate the total interest due over the loan duration
+        uint256 totalInterest = loanTerms.principal * loanTerms.durationSecs * loanTerms.interestRate
+            / (Constants.BASIS_POINTS_DENOMINATOR * Constants.SECONDS_IN_YEAR);
+
+        // verify interest amounts
+        if(totalInterest * stirfryData.payableToVaultedCurrencyRatio != stirfryData.borrowerVaultedCurrencyAmount)
+            revert OCS_InvalidInterestAmounts(
+                totalInterest,
+                stirfryData.payableToVaultedCurrencyRatio,
+                stirfryData.borrowerVaultedCurrencyAmount
+            );
+    }
+
+    /**
+     * @notice Perform loan initialization. Pull fixed interest amount from the borrower and add to vault.
+     *         Take custody of collateral form the lender. Tell LoanCore to create and start a loan.
+     *
+     * @dev The only collateral accepted by the STIRFRY Origination Controller is a Vault Factory vault.
+     *
+     * @param loanTerms                     The terms agreed by the lender and borrower.
+     * @param stirfryData                   The stirfry data to initialize the loan with.
+     * @param vaultAddress                  The address of the vault.
+     * @param borrower                      Address of the borrower.
+     * @param lender                        Address of the lender.
+     *
+     * @return loanId                       The unique ID of the new loan.
+     */
+    function _initialize(
+        LoanLibrary.LoanTerms calldata loanTerms,
+        StirfryData calldata stirfryData,
+        address vaultAddress,
+        address borrower,
+        address lender
+    ) internal nonReentrant returns (uint256 loanId) {
+        // get fee snapshot from fee controller
+        (LoanLibrary.FeeSnapshot memory feeSnapshot) = feeController.getFeeSnapshot();
+
+        // transfer fixed interest amount from borrower to lender's vault
+        IERC20(stirfryData.vaultedCurrency).safeTransferFrom(
+            borrower,
+            vaultAddress,
+            stirfryData.borrowerVaultedCurrencyAmount
```

Comment:
Could we just collect `totalInterest * stirfryData.payableToVaultedCurrencyRatio`?